File: ShoppingCart.Data/Models/Product.cs
using System;
using System.Collections.Generic;

namespace ShoppingCart.Data.Models
{
    public partial class Product
    {
        public Product()
        {
            OrderDetail = new HashSet<OrderDetail>();
        }

        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public string Url { get; set; }
        public int Quantity { get; set; }
        public string Specification { get; set; }
        public int CategoryId { get; set; }

        public virtual Category Category { get; set; }
        public virtual ICollection<OrderDetail> OrderDetail { get; set; }
    }
}

File: ShoppingCart.Web/ClientApp/src/app/Models/product.ts
export class Product {
  productId: number
  productName: string
  description: string
  price: number
  url: string
  quantity: number
  categoryId: number
  items: number
  specification: string
  specDetails: any[]

  constructor(id, name, description, price, imageUrl, categoryId, quantity, items, specification,specDetails) {
    this.productId = id
    this.productName = name
    this.description = description
    this.price = price
    this.url = imageUrl
    this.categoryId = categoryId
    this.quantity = quantity
    this.items = items
    this.specification = specification
    this.specDetails = specDetails
  }
}
